src/tests/test_x509_rpki.cpp

```diff
@@ -152,53 +161,136 @@ constexpr auto IPv6 = Botan::Cert_Extension::IPAddressBlocks::Version::IPv6;
+      result.test_eq("block 1 max is correct", block_1[0].max().value(), {200, 0, 0, 0});
+
+      result.test_eq("block 2 has correct safi", addr_blocks[2].safi(), std::optional<uint8_t>{2});
+      result.confirm(
+         "block 2 is inherited",
+         !std::get<IPAddressBlocks::IPAddressChoice<IPv4>>(addr_blocks[2].addr_choice()).ranges().has_value());
+
+      result.test_eq("block 3 has no safi", addr_blocks[3].safi(), std::optional<uint8_t>{std::nullopt});
+      const auto& block_3 =
+         std::get<IPAddressBlocks::IPAddressChoice<IPv6>>(addr_blocks[3].addr_choice()).ranges().value();
+
+      result.confirm("block 3 has correct size", block_3.size() == 1);
+      result.test_eq("block 3 min is correct",
+                     block_3[0].min().value(),
+                     {0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff});
+      result.test_eq("block 3 max is correct",
+                     block_3[0].max().value(),
+                     {0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff});
+
+      result.test_eq("block 24 has correct safi", addr_blocks[4].safi(), std::optional<uint8_t>{1});
+      result.confirm(
+         "block 4 is inherited",
+         !std::get<IPAddressBlocks::IPAddressChoice<IPv6>>(addr_blocks[4].addr_choice()).ranges().has_value());
+   }
+   {
+      const std::string filename("InvalidIPAddrBlocks.pem");
+      Botan::X509_Certificate cert(Test::data_file("x509/x509test/" + filename));
 
-   // 196.168.0.1
-   result.test_eq("ipv4 block 3 min", v4_blocks[3].min().value(), "C4A80001");
-   // 196.168.0.1
-   result.test_eq("ipv4 block 3 max", v4_blocks[3].max().value(), "C4A80001");
+      // cert contains the 10.0.32.0/20 prefix, but with a 9 for the unused bits
 
-   auto& v6_blocks = ipv6block.ranges().value();
+      result.confirm("extension is present", cert.v3_extensions().extension_set(IPAddressBlocks::static_oid()));
 
-   result.test_eq("ipv6 block 0 min", v6_blocks[0].min().value(), "FA800000000000000000000000000000");
-   result.test_eq("ipv6 block 0 max", v6_blocks[0].max().value(), "FA800000000000007FFFFFFFFFFFFFFF");
-   result.test_eq("ipv6 block 1 min", v6_blocks[1].min().value(), "FE200000000000000000000000000000");
-   result.test_eq("ipv6 block 1 max", v6_blocks[1].max().value(), "FE20000007FFFFFFFFFFFFFFFFFFFFFF");
-   result.test_eq("ipv6 block 2 min", v6_blocks[2].min().value(), "2003000068293435042010C5000000C4");
-   result.test_eq("ipv6 block 2 max", v6_blocks[2].max().value(), "2003000068293435042010C5000000C4");
-   result.test_eq("ipv6 block 3 min", v6_blocks[3].min().value(), "AB010000000000000000000000000001");
-   result.test_eq("ipv6 block 3 max", v6_blocks[3].max().value(), "CD020000000000000000000000000002");
+      auto ext = cert.v3_extensions().get_extension_object_as<IPAddressBlocks>();
+      result.confirm("extension is not decoded", ext == nullptr);
+   }
```

Comment:
This I find kind of sketchy btw, as in why does it not throw, the ext should be critical. I will have to look into this again.